File: frontend/service/actions/deployment.ts
"use server";

import { fetchFastApi } from "@/lib/fastapi";
import type { DeploymentUpdateStatus } from "@/service/types/deployment";

export async function getDeploymentUpdateStatus(): Promise<DeploymentUpdateStatus | null> {
    const response = await fetchFastApi("/deployment/update-status");
    if (!response.ok) {
        return null;
    }
    return (await response.json()) as DeploymentUpdateStatus;
}

File: backend/app/api/v1/deployment.py
from fastapi import APIRouter

from app.schemas.deployment import DeploymentUpdateStatusOut
from app.services.deployment_updates import get_deployment_update_status

router = APIRouter()


@router.get("/deployment/update-status", response_model=DeploymentUpdateStatusOut)
def deployment_update_status() -> DeploymentUpdateStatusOut:
    return get_deployment_update_status()
